CRM/Contribute/Form/UpdateSubscription.php

```diff
@@ -99,13 +97,9 @@ public function preProcess() {
       $this->assign('contactId', $this->_subscriptionDetails->contact_id);
     }
 
-    if (!CRM_Core_Permission::check('edit contributions')) {
-      if ($this->_subscriptionDetails->contact_id != $this->getContactID()) {
-        CRM_Core_Error::statusBounce(ts('You do not have permission to update subscription.'));
-      }
-      $this->_selfService = TRUE;
-    }
-    $this->assign('self_service', $this->_selfService);
+    $this->isSelfService();
+
+    $this->assign('self_service', $this->selfService);
```

Comment:
I think it's better now we have the function to keep calling it rather than the property ie 

```$this->isSelfService()```

rather than 

```
$this->selfService
```

in each place.

It might be we should add an early return to that fn if you are concerned about multiple calls to it

Reply:
Alright updated, the function does return early doing an is_null($this->selfService) where the default for the varible is NULL but then gets set to either TRUE or FALSE so seems ok to just constantly call the function